### ISSUE TITLE
FNX-4928 ⁃ Accessibility options under settings

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#1286]
### User Story
As a user, I want to be able to change my browser accessibility settings so I can browse comfortably.

### Dependencies
A-C Issue: Using system font for non-mobile pages (Landed in A-C) #577
UX designs: #1285 

Updated settings screen: https://mozilla.invisionapp.com/share/X3R7SA84BFR
New accessibility screen: https://mozilla.invisionapp.com/share/87RPXG0QDUZ

### Acceptance Criteria
I can adjust the size of website content font on desktop sites.

**Comment by vesta0** [comment]
@apbitner Is this blocked on https://bugzilla.mozilla.org/show_bug.cgi?id=1540615?

**Comment by apbitner** [comment]
@vesta0 Not that I'm aware of. In the UX ticket you mentioned 'Always enable zoom' would be a separate ticket for post-MVP, right?

**Comment by vesta0** [comment]
@apbitner you're right. Here it the post-mvp issue: https://github.com/mozilla-mobile/fenix/issues/2477

So for MVP, the only accessibility feature is the slider, correct?

**Comment by apbitner** [comment]
Correct

**Comment by ekager** [comment]
Hey @apbitner what are the minimum and maximum values here? I started implementing it going from 1% - 200% based on the midpoint being 100% but there's no way anyone would want it so small. Could you give me some direction? 

Also I was thinking we should maybe have a "reset to default value" or an "font adjustment" on/off switch to give users a quick way to undo this setting if it makes things wonky.

![text](https://user-images.githubusercontent.com/25442310/57816700-289d2180-7731-11e9-9841-b346c414925c.gif)

**Comment by apbitner** [comment]
Good question. We're just going to follow what most other browsers do, which is a scale from 50% to 200%, in increments of 5%. I think the increments will make it easier to get back to 100% as well.

The background is Violet 05 (#E7DFFF), 100% opacity. Text color is #232749.

**Comment by softvision-miralobontiu** [comment]
The scale doesn't apply on menus, or websites with/without Desktop site on.
Tested on the latest build 1.0.1920 (#11391807), with Samsung Galaxy S8 (Android 8), and Sony Xperia Z5 Premium (Android 7.1.1).

**Comment by ekager** [comment]
@softvision-miralobontiu sorry for the delay! I should have left a comment here, but I was waiting on an AC issue that was merged yesterday to be able to dynamically change these engine settings. #2747 will reload the current session so you will be able to see the changes immediately when updating the setting.

**Comment by softvision-miralobontiu** [comment]
Verified that the font size on the websites is changed according to the scale modified in Settings - Accessibility.
Tested on the latest build 1.0.1921 (#11430614), with OnePlus 5T (Android 9), Huawei P9 Lite (Android 6), Sony Xperia Z2 tablet (Android 6.0.1).

**Comment by buttercookie42** [comment]
> 2a. How does setAutomaticFontSizeAdjustment() work in with these new settings?

The currently implemented behaviour isn't ideal in any case, because if 100 % actually means "automatic font size adjustment enabled", then 100 % isn't necessarily 100 %, but instead whatever the OS-level font size setting is. So if the OS-level setting is e.g. at 125 %, then moving the Fennec slider from 100 % to 110 % actually **decreases** the font size, because in practice you're going from 125 % to 110 %.
Plus behind the scenes you need to manually need to turn on font inflation to keep it enabled (which apparently so far you didn't), because otherwise the effective font size on non-mobile-friendly pages decreases _even more_.

Basically I think that if you want some amount of manual control over the amount of font size scaling, you should decide between one of the following:
- complete manual control and never follow the OS-level setting
- a toggle switch for automatic font size setting, which when toggled would show/enable the manual control
- what Chrome does: the manual control is applied as an **additional multiplier** on top of the OS-level font size setting, in which case that would require some additional work on the GeckoView-side.

**Comment by ekager** [comment]
Thanks @buttercookie42 for the explanation! I think the last option is the most ideal going forward to follow Chrome's implementation of applying an additional multiplier. For now I will add a toggle using the system's text sizing.

**Comment by buttercookie42** [comment]
CC me when you file a corresponding GeckoView bug, and I'm happy to look at it.
